CollisionOperator/CollisionProcessors/neutronCEstd_class.f90

```diff
@@ -294,19 +308,28 @@ subroutine elastic(self, p, tally, collDat, thisCycle, nextCycle)
     logical(defBool)                       :: isFixed, hasDBRC
     character(100),parameter :: Here = 'elastic (neutronCEstd_class.f90)'
 
+    ! Assess if thermal scattering data is needed or not
+    if (self % nuc % needsSabEl(p % E)) collDat % MT = N_N_ThermEL
+
     ! Get reaction
     reac => uncorrelatedReactionCE_CptrCast( self % xsData % getReaction(collDat % MT, collDat % nucIdx))
-    if(.not.associated(reac)) call fatalError(Here,'Failed to get elastic neutron scatter')
+    if (.not.associated(reac)) call fatalError(Here,'Failed to get elastic neutron scatter')
 
     ! Scatter particle
     collDat % A =  self % nuc % getMass()
-    collDat % kT = self % nuc % getkT()
+
+    ! Retrieve kT from either material or nuclide
```

Comment:
This is mostly about my ignorance and probably not something to address now...
Is it conceivable to include only some nuclides in TMS in a given material? E.g., we could just rule out doing TMS on hydrogen if so.

Reply:
It would be possible I think! One could store some flag in the nuclide and use the normal xs instead of the temperature majorant for that one